server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsIndices.java

```diff
@@ -68,14 +71,58 @@ public class ClusterStatsIndices implements ToXContentFragment {
     private MappingStats mappings;
 
     public ClusterStatsIndices(List<ClusterStatsNodeResponse> nodeResponses, MappingStats mappingStats, AnalysisStats analysisStats) {
+        this(Set.of(IndexMetric.values()), nodeResponses, mappingStats, analysisStats);
+
+    }
+
+    public ClusterStatsIndices(
+        Set<IndexMetric> indicesMetrics,
+        List<ClusterStatsNodeResponse> nodeResponses,
+        MappingStats mappingStats,
+        AnalysisStats analysisStats
+    ) {
         Map<String, ShardStats> countsPerIndex = new HashMap<>();
+        this.docs = indicesMetrics.contains(IndexMetric.DOCS) ? new DocsStats() : null;
+        Consumer<DocsStats> docsStatsConsumer = (docs) -> {
```

Comment:
Consumer can be removed and this can be inline with the for loop

Reply:
Removed.